+sw_tests/+unit_tests/unittest_spinw_genlattice.m

```diff
@@ -0,0 +1,212 @@
+classdef unittest_spinw_genlattice < sw_tests.unit_tests.unittest_super
+
+    properties
+        swobj = [];
+        default_latt = struct('angle', repmat(pi/2, 1, 3), ...
+            'lat_const', repmat(3, 1, 3),  ...
+            'sym', zeros(3, 4, 0), ...
+            'origin', zeros(1, 3), 'label', 'P 0');
+        P2_sym = cat(3, [1 0 0 0; 0 1 0 0; 0 0 1 0], ...
+                        [-1 0 0 0; 0 1 0 0; 0 0 -1 0])
+    end
+    properties (TestParameter)
+        param_name = {'angle', 'lat_const'};
+        spgr = {'P 2', 3};  % spacegroup and index in symmetry.dat
+        invalid_spgr = {'P2', 'P 7', '-x,y,-x', '-x,y', eye(2)};
+        sym_param_name = {'sym', 'spgr'};
+        basis_vecs =  {[1/2 1/2 0; 0 1/2 1/2; 1/2 0 1/2], ... % RH
+            [1/2 1/2 0; 1/2 0 1/2; 0 1/2 1/2]}; % LH
+        nelem = {1,3}; % length of cell input for spgr
+        invalid_perm = {[1,4,2],[0,1,2], [1,1,1], 'bad', 'zzz', 'aaa', ...
+            {1,2,3}}
+        invalid_origin = {[-0.5,0,0], [0,2,0]};
+        invalid_label = {1, {'label'}}
+    end
+    
+    methods (TestMethodSetup)
+        function setup_spinw_model(testCase)
+            testCase.swobj = spinw(); % default init
+        end
+    end
+
+    methods (Test)     
+
+        function test_params_alter_lattice_fields_same_name(testCase, param_name)
+            value = [0.5, 0.5, 0.5];
+            testCase.swobj.genlattice(param_name, value);
+            expected_latt = testCase.default_latt;
+            expected_latt.(param_name) = value;
+            testCase.verify_val(expected_latt, testCase.swobj.lattice)
+        end
+        
+        function test_angled_degree_conversion(testCase)
+            testCase.swobj.genlattice('angled', [90, 90, 90]);
+            testCase.verify_val(testCase.default_latt, ...
+                testCase.swobj.lattice)
+        end
+        
+        function test_angle_and_angled_provided(testCase)
+            value = [1,1,1];
+            testCase.swobj.genlattice('angled',[1,1,1], 'angle', [1,1,1]);
+            expected_latt = testCase.default_latt;
+            expected_latt.angle = deg2rad(value);  % 'angled' used
```

Comment:
Good idea to do this, normally I could check whether a parameter has been set using the `params` returned from `sw_readparam` but because the default init of the `spinw.lattice` struct has values for the angle already,  to check whether 'angle' has been provided I would need to check against the default values - i.e. you could still fool it by passing the default angles to `angle` in addition to `angled` - so in this instance I will search for the name-value pair in `varargin`!